src/libsyntax/tokenstream.rs

```diff
@@ -156,15 +160,22 @@ impl TokenTree {
 #[derive(Clone, Debug)]
 pub enum TokenStream {
     Empty,
-    Tree(TokenTree),
-    JointTree(TokenTree),
+    Tree(TokenTree, IsJoint),
```

Comment:
I would've expected this would need `IsJoint` to be before `TokenTree` to make sure the smaller enum (`IsJoint`) ends up in the padding and does not increase the overall size.

I think `Tree` and `JointTree` as separate variants was a premature optimization, probably lack of awareness of the possibility of putting data in the padding (which exists thanks to the `Stream` variant with a `Lrc` field, regardless of other variants).

Reply:
The compiler reorders the fields:
```
print-type-size     variant `Tree`: 31 bytes
print-type-size         field `.1`: 1 bytes
print-type-size         padding: 6 bytes
print-type-size         field `.0`: 24 bytes, alignment: 8 bytes
```
I recently added an assertion on the size of `TokenStream`, so if this change had increased the size it would have been caught.